design/20220722-publicbundle.md

```diff
@@ -0,0 +1,326 @@
+and [Fedora's equivalent](https://src.fedoraproject.org/rpms/ca-certificates), as background on
+how popular Linux distributions package their default certificate bundles.
+
+The ideal situation for the vast majority of applications is likely to be that they don't have to
+explicitly configure their application's trust bundle.
+
+By way of example, a Linux Golang application would [currently](https://github.com/golang/go/blob/2d655fb15a50036547a6bf8f77608aab9fb31368/src/crypto/x509/root_linux.go#L8-L15)
+look in the following locations by default:
+
+```go
+var certFiles = []string{
+	"/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
+	"/etc/pki/tls/certs/ca-bundle.crt",                  // Fedora/RHEL 6
+	"/etc/ssl/ca-bundle.pem",                            // OpenSUSE
+	"/etc/pki/tls/cacert.pem",                           // OpenELEC
+	"/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", // CentOS/RHEL 7
+	"/etc/ssl/cert.pem",                                 // Alpine Linux
+}
+```
+
+It's a chore to manually specify a location for a trust store, and it's unfamiliar and error-prone
+for many users. However, if a user instead overwrites any of these files with their own
+`trust`-generated certificate trust bundle they'll currently overwrite their public trust bundle
+provided by their OS / container base image.
+
+Unless the user's application only talks to services which the user also controls, what users are
+likely to really want in most cases is a bundle containing both the publicly trusted certs _and_
+the user's own roots.
+
+It's possible in most distros to append to the cert bundle, but that usually involves running some
+command to generate the final bundle, which would change how the user invokes their application
+inside their container.
+
+Ideally, the whole bundle a user requires would simply be available at one of the above locations.
+
+### Goals
+
+- A user configuring a `Bundle` can add a public trust bundle at least as easily as they can add certs from a `ConfigMap` or `Secret`
+- It's clear to users exactly what version of the public bundle is used at any given time, for auditing purposes
+- A user whose application must use both public and private CAs can confidently overwrite their "system" trust store in a container using a trust `Bundle`
+- A user can update any bundle in use with any version of the trust controller and be confident that it will work
+
+### Non-Goals
+
+- Editing/pruning the public trust bundle (out of scope for the initial feature but might be desirable later)
+- Supporting [OS-specific](https://superuser.com/questions/437330/how-do-you-add-a-certificate-authority-ca-to-ubuntu) certificate update mechanisms
+- Supporting a plethora of options for sources of public trust bundles; we might want this in the future but we should start with one
+
+## Proposal
+
+We propose that the cert-manager project becomes a distributor of its own public trust bundle, which at
```

Comment:
I am a very strong -1 on this goal. I don't think we have staffing nor the desire to take on the burden of ensuring this is up to date, notifying users of changes, etc etc. cert-manager has always been a "middle man" and this puts us in an extremely privileged position in many people's infrastructure.

Reply:
🤷 We end up effectively being a distributor of a public trust bundle in basically every proposed design so far; I'm just acknowledging it upfront.

I think we can very easily automate all of this work

>  this puts us in an extremely privileged position in many people's infrastructure.

We're in an extremely privileged position in every single k8s cluster that `trust` or cert-manager is deployed into already.